wingtips-core/src/main/java/com/nike/wingtips/Span.java

```diff
@@ -430,118 +391,12 @@ public static Span fromKeyValueString(String keyValueStr) {
     public String toJSON() {
         // Profiling shows this JSON creation to generate a lot of garbage in certain situations, so we should cache the result.
         if (cachedJsonRepresentation == null)
-            cachedJsonRepresentation = calculateJson();
+            cachedJsonRepresentation = SpanParser.convertSpanToJSON(this);
 
         return cachedJsonRepresentation;
     }
 
-    /**
-     * Calculates and returns the JSON representation of this span instance. We build this manually ourselves to avoid pulling in an extra dependency
-     * (e.g. Jackson) just for building a simple JSON string.
-     */
-    protected String calculateJson() {
-        StringBuilder builder = new StringBuilder();
-
-        builder.append("{\"").append(TRACE_ID_FIELD).append("\":\"").append(traceId);
-        builder.append("\",\"").append(PARENT_SPAN_ID_FIELD).append("\":\"").append(parentSpanId);
-        builder.append("\",\"").append(SPAN_ID_FIELD).append("\":\"").append(spanId);
-        builder.append("\",\"").append(SPAN_NAME_FIELD).append("\":\"").append(spanName);
-        builder.append("\",\"").append(SAMPLEABLE_FIELD).append("\":\"").append(sampleable);
-        builder.append("\",\"").append(USER_ID_FIELD).append("\":\"").append(userId);
-        builder.append("\",\"").append(SPAN_PURPOSE_FIELD).append("\":\"").append(spanPurpose.name());
-        builder.append("\",\"").append(START_TIME_EPOCH_MICROS_FIELD).append("\":\"").append(spanStartTimeEpochMicros);
-        if (isCompleted()) {
-            builder.append("\",\"").append(DURATION_NANOS_FIELD).append("\":\"").append(durationNanos);
-        }
-        builder.append("\"}");
-
-        return builder.toString();
-    }
-
-    /**
-     * @return The {@link Span} represented by the given JSON string, or null if a proper span could not be deserialized from the given string.
-     *          <b>WARNING:</b> This method assumes the JSON you're trying to deserialize originally came from {@link #toJSON()}.
-     *          This assumption allows it to be as fast as possible, not have to check for malformed JSON, not worry about syntactically-correct-but-annoying-to-deal-with whitespace,
-     *          not have to use a third party utility like Jackson, etc.
-     */
-    public static Span fromJSON(String json) {
-        try {
```

Comment:
In order to prevent breaking changes, the public static `fromJSON(String)` and `fromKeyValueString(String)` methods should not be removed entirely, but instead simply call the new methods in `SpanParser`.